packages/nodes-base/nodes/Microsoft/OneDrive/TriggerDescription.ts

```diff
@@ -0,0 +1,289 @@
+import type { INodeProperties } from 'n8n-workflow';
+import { MICROSOFT_DRIVE_FILE_URL_REGEX, MICROSOFT_DRIVE_FOLDER_URL_REGEX } from './constants';
+
+export const fileRLC: INodeProperties = {
+	displayName: 'File',
+	name: 'fileId',
+	type: 'resourceLocator',
+	default: { mode: 'id', value: '' },
+	required: true,
+	modes: [
+		{
+			displayName: 'Link',
+			name: 'url',
+			type: 'string',
+			placeholder:
+				'e.g. https://onedrive.live.com/edit.aspx?resid=170B5C65E30736A3!257&cid=170b5c65e30736a3&CT=1708697995542&OR=ItemsView',
+			extractValue: {
+				type: 'regex',
+				regex: MICROSOFT_DRIVE_FILE_URL_REGEX,
+			},
+			validation: [
+				{
+					type: 'regex',
+					properties: {
+						regex: MICROSOFT_DRIVE_FILE_URL_REGEX,
+						errorMessage: 'Not a valid Microsoft Drive File URL',
+					},
+				},
+			],
+		},
+		{
+			displayName: 'ID',
+			name: 'id',
+			type: 'string',
+			placeholder: 'e.g. 170B5C65E30736A3!257',
+			validation: [
+				{
+					type: 'regex',
+					properties: {
+						regex: '[a-zA-Z0-9\\!%21]{5,}',
+						errorMessage: 'Not a valid Microsoft Drive File ID',
+					},
+				},
+			],
+			url: '=https://onedrive.live.com/...',
+		},
+	],
+	description:
+		"The file to operate on. The 'By URL' option only accepts URLs that start with 'https://onedrive.live.com'.",
+};
+
+export const folderRLC: INodeProperties = {
+	displayName: 'Folder',
+	name: 'folderId',
+	type: 'resourceLocator',
+	default: { mode: 'id', value: '', cachedResultName: '' },
+	required: true,
+	modes: [
+		{
+			displayName: 'Link',
+			name: 'url',
+			type: 'string',
+			placeholder:
+				'e.g. https://onedrive.live.com/?WT.mc_id=PROD%5FOL%2DWeb%5FInApp%5FLeftNav%5FFreeOfficeBarOD&ocid=PROD%5FOL%2DWeb%5FInApp%5FLeftNav%5FFreeOfficeBarOD&id=170B5C65E30736A3%21136&cid=170B5C65E30736A3',
```

Comment:
For me simpler urls like `https://onedrive.live.com/?id=4A4A05756B39BA4A%2112047&cid=4A4A05756B39BA4A` seem are the default.